provider/bigquery.go

```diff
@@ -42,67 +44,28 @@ const (
 	sleepTime = 1 * time.Second
 )
 
-type BQOfflineStoreConfig struct {
-	Config       pc.SerializedConfig
-	ProjectId    string
-	ProviderType p_type.Type
-	QueryImpl    BQOfflineTableQueries
-	logger       logging.Logger
-}
-
-type BQOfflineTableQueries interface {
-	tableExists(tableName string) string
-	viewExists(viewName string) string
-	determineColumnType(valueType types.ValueType) (string, error)
-	primaryTableCreate(name string, columnString string) string
-	upsertQuery(tb string, columns string, placeholder string) string
-	createValuePlaceholderString(columns []TableColumn) string
-	registerResources(client *bigquery.Client, tableName string, schema ResourceSchema, timestamp bool) error
-	writeUpdate(table string) string
-	writeInserts(table string) string
-	writeExists(table string) string
-	newBQOfflineTable(name string, columnType string) string
-	materializationCreate(tableName string, resultName string) string
-	materializationIterateSegment(tableName string, start int64, end int64) string
-	getNumRowsQuery(tableName string) string
-	getTablePrefix() string
-	setTablePrefix(prefix string)
-	getContext() context.Context
-	setContext()
-	castTableItemType(v interface{}, t interface{}) interface{}
-	materializationExists(tableName string) string
-	materializationDrop(tableName string) string
-	materializationUpdate(client *bigquery.Client, tableName string, sourceName string) error
-	monitorJob(job *bigquery.Job) error
-	transformationCreate(name string, query string) string
-	getColumns(client *bigquery.Client, name string) ([]TableColumn, error)
-	transformationUpdate(client *bigquery.Client, tableName string, query string) error
-	trainingSetCreate(store *bqOfflineStore, def TrainingSetDef, tableName string, labelName string) error
-	trainingSetUpdate(store *bqOfflineStore, def TrainingSetDef, tableName string, labelName string) error
-	trainingSetQuery(store *bqOfflineStore, def TrainingSetDef, tableName string, labelName string, isUpdate bool) error
-	atomicUpdate(client *bigquery.Client, tableName string, tempName string, query string) error
-	trainingRowSelect(columns string, trainingSetName string) string
-	primaryTableRegister(tableName string, sourceName string) string
-	getTableName(tableName string) string
-}
-
 type defaultBQQueries struct {
-	TablePrefix string
-	Ctx         context.Context
+	ProjectId string
+	DatasetId string
+	Ctx       context.Context
```

Comment:
Don't like ctx/logger being passed in like this unless this gets re-created each time?